core/src/main/java/org/apache/james/mime4j/MimeStreamParser.java

```diff
@@ -0,0 +1,122 @@
+/****************************************************************
```

Comment:
How do you know where to draw the interface with only one implementation.

I didn’t understand “private packages” because you can’t have a usable package where everything is private.

Single class interface implementations are meaningless - every change to the interface will be a breaking change.